spec/features/users/recovery_code_spec.rb

```diff
@@ -48,6 +50,62 @@
       it_behaves_like 'recovery code page'
     end
   end
+
+  context 'with javascript enabled', js: true do
+    let(:invisible_selector) { generate_class_selector('invisible') }
+    let(:accordion_control_selector) { generate_class_selector('accordion-header-control') }
+
+    it 'prompts the user to enter their personal key to confirm they have it' do
+      sign_in_and_2fa_user
+      click_link t('profile.links.regenerate_recovery_code')
+
+      expect_accordion_content_to_be_hidden_by_default
+
+      expand_accordion
+
+      expect_accordion_content_to_become_visible
+
+      click_acknowledge_recovery_code
+
+      expect_confirmation_modal_to_appear_with_first_code_field_in_focus
+
+      press_shift_tab
+
+      expect_back_button_to_be_in_focus
+
+      click_back_button
+
+      expect_to_be_back_on_manage_recovery_code_page_with_continue_button_in_focus
+
+      click_acknowledge_recovery_code
+
```

Comment:
seems like the pattern you're following here is newlines between exercise and expectation steps--- so we can remove this newline?